src/runtime/transaction_execution.zig

```diff
@@ -445,6 +450,7 @@ pub fn executeTransaction(
             }
             break;
         };
+        instruction_index += 1;
```

Comment:
Suggestion:
```zig
const maybe_instruction_fail: ?struct { err: InstructionError, index: usize } =
    for (transaction.instruction_infos, 0..) |instruction_info, i| {
        executor.executeInstruction(
            allocator,
            &tc,
            instruction_info,
        ) catch |err| switch (err) {
            error.OutOfMemory => return error.OutOfMemory,
            else => |e| break .{ .err = e, .index = i },
        };
    } else null;
```
In combination with my other suggestion about tying together the nullability of `err` and `err_index`.